include/dump_offload.hpp

```diff
@@ -0,0 +1,443 @@
+        });
+    }
+
+    /**
+     * @brief  Invokes InitiateOffload method of dump manager which
+     *         directs dump manager to start writing on unix domain socket.
+     *
+     * @return void
+     */
+    void initiateOffload()
+    {
+        crow::connections::systemBus->async_method_call(
+            [this,
+             self(shared_from_this())](const boost::system::error_code ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error: {}", ec);
+                if (ec.value() == EBADR)
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::not_found);
+                }
+                else
+                {
+                    this->connection->sendStreamErrorStatus(
+                        boost::beast::http::status::internal_server_error);
+                }
+                this->connection->close();
+                this->cleanupSocketFiles();
+                return;
+            }
+        },
+            "xyz.openbmc_project.Dump.Manager",
+            "/xyz/openbmc_project/dump/" + dumpType + "/entry/" + entryID,
+            "xyz.openbmc_project.Dump.Entry", "InitiateOffload",
+            unixSocketPath.c_str());
+    }
+
+    /**
+     * @brief  This function setup a timer for retrying unix socket connect.
+     *
+     * @return void
+     */
+    void retrySocketConnect()
+    {
+        waitTimer.expires_after(std::chrono::milliseconds(1000));
+
+        waitTimer.async_wait([this, self(shared_from_this())](
+                                 const boost::system::error_code& ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("Async_wait failed {}", ec);
+                return;
+            }
+
+            if (connectRetryCount < maxConnectRetryCount)
+            {
+                BMCWEB_LOG_ERROR("Failed to connect, reached max retry count: ",
+                                 connectRetryCount);
+                connectRetryCount++;
+                doConnect();
+            }
+            else
+            {
+                BMCWEB_LOG_ERROR("Failed to connect, reached max retry count: ",
+                                 connectRetryCount);
+                waitTimer.cancel();
+                this->cleanupSocketFiles();
+                this->connection->setStreamHeaders("Retry-After", "60");
+                this->connection->sendStreamErrorStatus(
+                    boost::beast::http::status::service_unavailable);
+                this->connection->close();
+                return;
+            }
+        });
+    }
+
+    void resetOffloadURI()
+    {
+        std::string value;
+        crow::connections::systemBus->async_method_call(
+            [this,
+             self(shared_from_this())](const boost::system::error_code ec) {
+            if (ec)
+            {
+                BMCWEB_LOG_ERROR("DBUS response error: Unable to set "
```

Comment:
not_found is a user error, could we have a warning trace there but a this error trace for the internal server error ?

Reply:
resetOffloadURI() method is internal function which gets invoked after dump offload completes.
user does not aware of this internal details of these.
for any reason if dump manager app is down then this not_found case hit only.

Reply:
not_found is a user error. 


            if (ec)
            {
               if (ec.value() == EBADR)
                {
                    this->connection->sendStreamErrorStatus(
                        boost::beast::http::status::not_found);
                }
                else
                {
                BMCWEB_LOG_ERROR("DBUS response error: Unable to set "
                                                 "the dump OffloadUri {}",
                                 ec);
                    this->connection->sendStreamErrorStatus(
                        boost::beast::http::status::internal_server_error);
                }
                return;

Reply:
ok understand your point. looks like its better to return internal error as you have suggested @gtmills

Reply:
done

Reply:
It looks like you just removed the 
```
if (ec.value() == EBADR) { 
this->connection->sendStreamErrorStatus( boost::beast::http::status::not_found); 
}
```

I would think we still want the not_found?

Reply:
anyways not_found is not helpful here.
any D-bus error would be some internal D-bus app error.
so internal error would be better for client

Reply:
removed error trace for found_error